src/main/java/com/kodedu/service/impl/FileWatchServiceImpl.java

```diff
@@ -118,6 +120,7 @@ private void watchPathChanges() {
             }
 
             List<WatchEvent<?>> watchEvents = watchKey.pollEvents();
```

Comment:
pollEvents is not blocking, might want to add a sleep somewhere in here or use a self-scheduling method instead of a loop